cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaMirrorMaker2AssemblyOperator.java

```diff
@@ -432,6 +435,27 @@ private Future<Void> removeAnnotation(Reconciliation reconciliation, KafkaMirror
                 .compose(ignored -> Future.succeededFuture());
     }
 
+    /**
+     * Patches the KafkaMirrorMaker2 CR to remove the supplied annotations.
+     *
+     * @param reconciliation    Reconciliation marker
+     * @param resource          KafkaMirrorMaker2 resource from which the annotations should be removed
+     * @param annotationKeys     List of Annotations that should be removed
+     *
+     * @return  Future that indicates the operation completion
+     */
+    private Future<Void> removeAnnotations(Reconciliation reconciliation, KafkaMirrorMaker2 resource, List<String> annotationKeys) {
+        LOGGER.debugCr(reconciliation, "Removing annotations {}", annotationKeys);
+        Map<String, String> annotationsToRemove = annotationKeys.stream().collect(Collectors.toMap(key -> key, key -> ""));
+        KafkaMirrorMaker2 patchedKafkaMirrorMaker2 = new KafkaMirrorMaker2Builder(resource)
+                .editMetadata()
+                .removeFromAnnotations(annotationsToRemove)
+                .endMetadata()
+                .build();
+        return resourceOperator.patchAsync(reconciliation, patchedKafkaMirrorMaker2)
+                .compose(ignored -> Future.succeededFuture());
```

Comment:
Why is this needed?

Reply:
I matched the existing behaviour that is present when removing a single annotation where the future just completes whether it succeeded or failed.

Reply:
And does that really work? I think the `.compose` will not be called if the patch fails. If you wanna suppress the error, the best thing to do is I think to use `recover`. E.g. https://github.com/strimzi/strimzi-kafka-operator/blob/8e30be30de02118ab43e12f45304039419d55ff3/cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaReconciler.java#L377

Reply:
Ah, no it isn't doing that, I think it's actually just mapping the result, the javadoc threw me, I'll change it to `mapEmpty()`